libs/packages/components/src/lib/popup/popup.directive.ts

```diff
@@ -0,0 +1,35 @@
+import { Directive, Input, ElementRef, Renderer2, AfterViewInit } from '@angular/core';
+
+
+@Directive({
+  selector: '[sdsPopup]',
+  exportAs: 'sdsPopup'
+})
+export class SdsPopupDirective implements AfterViewInit {
+  @Input() position: string;
+  @Input() place: string;
+  @Input() className: string;
+  getClassNameArray = [];
+  tooltipContent: any;
+  tooltipConfig: HTMLElement;
+  sdsPopupDiv: HTMLElement;
+
+  constructor(private el: ElementRef, private renderer: Renderer2) {
+    this.renderer.addClass(this.el.nativeElement, 'sds-popup');
+    this.sdsPopupDiv = document.createElement('div');
+    this.renderer.addClass(this.sdsPopupDiv, 'sds-popup__content');
+  }
+
+
+  ngAfterViewInit() {
+    this.tooltipConfig = this.el.nativeElement.querySelector('.tooltipContent');
+    this.renderer.addClass(this.sdsPopupDiv, this.place);
+    this.renderer.addClass(this.sdsPopupDiv, this.position);
+    this.renderer.appendChild(this.sdsPopupDiv, this.tooltipConfig);
+    this.renderer.appendChild(this.el.nativeElement, this.sdsPopupDiv);
+    const strinx = this.className.split(',');
+    for (let i = 0; i < strinx.length; i++) {
+      this.renderer.addClass(this.tooltipConfig, strinx[i]);
+    }
```

Comment:
We should just let the target html element have all the styling, we should only position it correctly on the page.